### PR TITLE
More explicit relationship between values and properties

### PR DESCRIPTION
**Pull request by elsom25**
- [x] Add Value#primary_property
- [x] Change txt serializations to leverage this
- [x] Update `attributes.yml` to have normative IDs

As for output for `attribute_values.txt`, I went with:
- `# Format: {GID} : {Name} [{Attribute Name}]`

Considered these as well. `>` was tempting but leads to hard-to-read files with mis-alignment:
- `# Format: {GID} : {Attribute name} > {Value name}`
- `# Format: {GID} : {Name} [{All attribute names separated by ;}]`

<details><summary>Wood with primary only</summary>
<pre>
gid://shopify/TaxonomyValue/67    : Wood [Material]
gid://shopify/TaxonomyValue/309   : Wood [Decoration material]
gid://shopify/TaxonomyValue/490   : Wood [Door material]
gid://shopify/TaxonomyValue/521   : Wood [Food storage material]
gid://shopify/TaxonomyValue/569   : Wood [Habitat material]
gid://shopify/TaxonomyValue/889   : Wood [Clothing accessory material]
gid://shopify/TaxonomyValue/958   : Wood [Jewelry material]
gid://shopify/TaxonomyValue/978   : Wood [Bristle material]
gid://shopify/TaxonomyValue/1491  : Wood [Fastener material]
gid://shopify/TaxonomyValue/1551  : Wood [Suitable for crafting material]
gid://shopify/TaxonomyValue/1628  : Wood [Wreath design]
gid://shopify/TaxonomyValue/2067  : Wood [Blade material]
gid://shopify/TaxonomyValue/2097  : Wood [Handle material]
gid://shopify/TaxonomyValue/2220  : Wood [Brush material]
gid://shopify/TaxonomyValue/2750  : Wood [Scale model material]
gid://shopify/TaxonomyValue/2906  : Wood [Instrument accessory material]
gid://shopify/TaxonomyValue/3134  : Wood [Drum hardware material]
gid://shopify/TaxonomyValue/3150  : Wood [Mount material]
gid://shopify/TaxonomyValue/3198  : Wood [Mallet material]
gid://shopify/TaxonomyValue/3252  : Wood [Instrument part material]
gid://shopify/TaxonomyValue/3392  : Wood [Instrument material]
gid://shopify/TaxonomyValue/3636  : Wood [Accessory material]
gid://shopify/TaxonomyValue/4301  : Wood [Shelf material]
gid://shopify/TaxonomyValue/7966  : Wood [Compatible filament]
gid://shopify/TaxonomyValue/9176  : Wood [Foundation material]
gid://shopify/TaxonomyValue/9188  : Wood [Slat material]
gid://shopify/TaxonomyValue/9377  : Wood [Suitable for material type]
gid://shopify/TaxonomyValue/9424  : Wood [Intended application]
gid://shopify/TaxonomyValue/9508  : Wood [Wall patching application]
gid://shopify/TaxonomyValue/9573  : Wood [Building board material]
gid://shopify/TaxonomyValue/9660  : Wood [Siding material]
gid://shopify/TaxonomyValue/9671  : Wood [Tile material]
gid://shopify/TaxonomyValue/9690  : Wood [Tile look]
gid://shopify/TaxonomyValue/10213 : Wood [Hammer head material]
gid://shopify/TaxonomyValue/11587 : Wood [Eyewear frame material]
gid://shopify/TaxonomyValue/11810 : Wood [Artwork frame material]
gid://shopify/TaxonomyValue/11934 : Wood [Window treatment material]
gid://shopify/TaxonomyValue/12255 : Wood [Barware material]
gid://shopify/TaxonomyValue/12453 : Wood [Tableware material]
gid://shopify/TaxonomyValue/12501 : Wood [Flatware material]
gid://shopify/TaxonomyValue/12582 : Wood [Plant support material]
gid://shopify/TaxonomyValue/13059 : Wood [Office supply material]
gid://shopify/TaxonomyValue/13127 : Wood [Office instrument material]
gid://shopify/TaxonomyValue/13193 : Wood [Frame design]
gid://shopify/TaxonomyValue/13711 : Wood [Bat material]
gid://shopify/TaxonomyValue/13787 : Wood [Stick material]
gid://shopify/TaxonomyValue/13855 : Wood [Racket material]
gid://shopify/TaxonomyValue/14372 : Wood [Butt cap material]
gid://shopify/TaxonomyValue/14393 : Wood [Rod material]
gid://shopify/TaxonomyValue/14457 : Wood [Club type]
gid://shopify/TaxonomyValue/14501 : Wood [Bow material]
gid://shopify/TaxonomyValue/14964 : Wood [Item material]
gid://shopify/TaxonomyValue/15339 : Wood [Toy/Game material]
gid://shopify/TaxonomyValue/15380 : Wood [Furniture/Fixture material]
gid://shopify/TaxonomyValue/15391 : Wood [Disposable/Reusable item material]
gid://shopify/TaxonomyValue/15436 : Wood [Bag/Case material]
gid://shopify/TaxonomyValue/15511 : Wood [Lumber/Wood type]
gid://shopify/TaxonomyValue/15568 : Wood [Embellishment/Trim material]
gid://shopify/TaxonomyValue/15634 : Wood [Art/Crafting tool material]
gid://shopify/TaxonomyValue/16456 : Wood [Pole/Post material]
gid://shopify/TaxonomyValue/16497 : Wood [Tool/Utensil material]
gid://shopify/TaxonomyValue/16518 : Wood [Book/File cover material]
gid://shopify/TaxonomyValue/16671 : Wood [Arrow/Bolt material]
</pre>
</details> 

<details><summary>Wood with all attributes</summary>
<pre>
gid://shopify/TaxonomyValue/67    : Wood [Material]
gid://shopify/TaxonomyValue/309   : Wood [Decoration material]
gid://shopify/TaxonomyValue/490   : Wood [Door material; Garage door material]
gid://shopify/TaxonomyValue/521   : Wood [Food storage material]
gid://shopify/TaxonomyValue/569   : Wood [Habitat material]
gid://shopify/TaxonomyValue/889   : Wood [Clothing accessory material]
gid://shopify/TaxonomyValue/958   : Wood [Jewelry material]
gid://shopify/TaxonomyValue/978   : Wood [Bristle material]
gid://shopify/TaxonomyValue/1491  : Wood [Fastener material]
gid://shopify/TaxonomyValue/1551  : Wood [Suitable for crafting material]
gid://shopify/TaxonomyValue/1628  : Wood [Wreath design]
gid://shopify/TaxonomyValue/2067  : Wood [Blade material; Jaw material; Scale material; Teeth material]
gid://shopify/TaxonomyValue/2097  : Wood [Handle material]
gid://shopify/TaxonomyValue/2220  : Wood [Brush material]
gid://shopify/TaxonomyValue/2750  : Wood [Scale model material]
gid://shopify/TaxonomyValue/2906  : Wood [Instrument accessory material]
gid://shopify/TaxonomyValue/3134  : Wood [Drum hardware material]
gid://shopify/TaxonomyValue/3150  : Wood [Mount material]
gid://shopify/TaxonomyValue/3198  : Wood [Mallet material]
gid://shopify/TaxonomyValue/3252  : Wood [Instrument part material; Fingerboard material; Frog material; Lip plate material; Neck material]
gid://shopify/TaxonomyValue/3392  : Wood [Instrument material; Instrument bow material]
gid://shopify/TaxonomyValue/3636  : Wood [Accessory material; 3D glasses material; Bottle material; Cocktail decoration material; Moving/Shipping box material; Needle housing material; Shaft material; Suction cup material; Tank material; Text plate material]
gid://shopify/TaxonomyValue/4301  : Wood [Shelf material]
gid://shopify/TaxonomyValue/7966  : Wood [Compatible filament]
gid://shopify/TaxonomyValue/9176  : Wood [Foundation material]
gid://shopify/TaxonomyValue/9188  : Wood [Slat material]
gid://shopify/TaxonomyValue/9377  : Wood [Suitable for material type]
gid://shopify/TaxonomyValue/9424  : Wood [Intended application]
gid://shopify/TaxonomyValue/9508  : Wood [Wall patching application]
gid://shopify/TaxonomyValue/9573  : Wood [Building board material]
gid://shopify/TaxonomyValue/9660  : Wood [Siding material]
gid://shopify/TaxonomyValue/9671  : Wood [Tile material]
gid://shopify/TaxonomyValue/9690  : Wood [Tile look]
gid://shopify/TaxonomyValue/10213 : Wood [Hammer head material]
gid://shopify/TaxonomyValue/11587 : Wood [Eyewear frame material]
gid://shopify/TaxonomyValue/11810 : Wood [Artwork frame material]
gid://shopify/TaxonomyValue/11934 : Wood [Window treatment material]
gid://shopify/TaxonomyValue/12255 : Wood [Barware material]
gid://shopify/TaxonomyValue/12453 : Wood [Tableware material]
gid://shopify/TaxonomyValue/12501 : Wood [Flatware material]
gid://shopify/TaxonomyValue/12582 : Wood [Plant support material]
gid://shopify/TaxonomyValue/13059 : Wood [Office supply material]
gid://shopify/TaxonomyValue/13127 : Wood [Office instrument material]
gid://shopify/TaxonomyValue/13193 : Wood [Frame design]
gid://shopify/TaxonomyValue/13711 : Wood [Bat material]
gid://shopify/TaxonomyValue/13787 : Wood [Stick material]
gid://shopify/TaxonomyValue/13855 : Wood [Racket material]
gid://shopify/TaxonomyValue/14372 : Wood [Butt cap material]
gid://shopify/TaxonomyValue/14393 : Wood [Rod material]
gid://shopify/TaxonomyValue/14457 : Wood [Club type]
gid://shopify/TaxonomyValue/14501 : Wood [Bow material; Core material; Riser material]
gid://shopify/TaxonomyValue/14964 : Wood [Item material]
gid://shopify/TaxonomyValue/15339 : Wood [Toy/Game material]
gid://shopify/TaxonomyValue/15380 : Wood [Furniture/Fixture material; Frame material; Housing material; Leg material; Tabletop material; Top material]
gid://shopify/TaxonomyValue/15391 : Wood [Disposable/Reusable item material; Disposable bag material; Disposable tableware material; Shopping bag material; Waste bag material]
gid://shopify/TaxonomyValue/15436 : Wood [Bag/Case material; Diaper bag material]
gid://shopify/TaxonomyValue/15511 : Wood [Lumber/Wood type]
gid://shopify/TaxonomyValue/15568 : Wood [Embellishment/Trim material]
gid://shopify/TaxonomyValue/15634 : Wood [Art/Crafting tool material]
gid://shopify/TaxonomyValue/16456 : Wood [Pole/Post material; Pole material; Post material]
gid://shopify/TaxonomyValue/16497 : Wood [Tool/Utensil material]
gid://shopify/TaxonomyValue/16518 : Wood [Book/File cover material; Book cover material; Folder/Report cover material]
gid://shopify/TaxonomyValue/16671 : Wood [Arrow/Bolt material; Arrow material; Bolt material]
</pre>
</details>

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -11,10 +11,6 @@ class Category < ApplicationRecord
   has_many :categories_properties, dependent: :destroy
   has_many :properties, through: :categories_properties, foreign_key: :property_friendly_id
 
-  def property_friendly_ids
```
Comment (KComrade53) [inline app/models/category.rb:14]: Didn't like these clean getters? 😅

File: test/integration/all_data_files_import_test.rb [hunk 2]
```diff
@@ -90,50 +93,32 @@ def before_all
...
-    @raw_attributes_data.each do |raw_attribute|
-      property = Property.find(raw_attribute.fetch("id"))
+    @raw_attributes_data.select { _1.key?("values") }.each do |raw_attribute|
+      values_via_raw_id = Property.find(raw_attribute.fetch("id")).property_values
+      values_via_raw_values = raw_attribute.fetch("values").map { PropertyValue.find_by(friendly_id: _1) }
 
-      next unless raw_attribute.fetch("values_from", nil).nil?
-
-      raw_value_friendly_ids = raw_attribute.fetch("values", [])
-      refute raw_value_friendly_ids.empty?
```
Comment (KComrade53) [inline test/integration/all_data_files_import_test.rb:108]: Can we add another test then to make sure this case is captured? We don't want any attributes that have an empty values array or it tells us something went wrong

**Review by KComrade53 — APPROVED** [review]

File: test/integration/all_data_files_import_test.rb (hunk 2 above)
Comment (elsom25) [reply test/integration/all_data_files_import_test.rb:108]: Ah thought this was just trying to error correct some parsing. An explicitly named test to that effect feels good

Comment (KComrade53) [reply test/integration/all_data_files_import_test.rb:108]: Added a test for it: https://github.com/Shopify/product-taxonomy/pull/129
